torch/_inductor/lowering.py

```diff
@@ -292,10 +292,6 @@ def wrapped(*args, **kwargs):
             unpacked = True
             args = args[0]
 
-        # explicitly assert for "out=" ops for better error messages
```

Comment:
Is the metadata mutated at all here ? I'm not sure we handle that, maybe we can check ?

Reply:
agreed - wonder what would be the best way to check it? (do we have something like `example_value` where we can e.g. compare the before vs. after strides for the `out` arg?)

Reply:
After running an `out kernel` you can go iterate through the `args_flat` and check they have the same metadata as the equivalent arg in `tensor_args`

https://github.com/pytorch/pytorch/blob/main/torch/_inductor/ir.py#L4171

Reply:
Thanks! added the check in https://github.com/pytorch/pytorch/pull/129422/files#diff-cf6ca00beddc32a2a6a2933fb9913b6a2b925ffc3b745488967210e4343134acR4328